parsing/lexer.mll

```diff
@@ -732,7 +741,16 @@ rule token = parse
   | "#"
       { if not (at_beginning_of_line lexbuf.lex_start_p)
         then HASH
-        else try directive Hash lexbuf with Failure _ -> HASH
+        else
+          (* Try to parse as #syntax directive first *)
+          let pos = lexbuf.lex_curr_pos in
+          let curpos = lexbuf.lex_curr_p in
+          try syntax_directive Hash lexbuf
+          with Failure _ ->
+            (* Reset position and try normal directive *)
```

Comment:
I'd rather we try to avoid this kind of backtracking if we can. Is it definitely necessary? The `directive` case doesn't bother with it. I notice that `syntax_directive` includes a catchall case, whereas `directive` doesn't, does that cause a difference? Can we just merge `syntax_directive` into `directive` rather than have them separate?

Reply:
I'll give it a go.